src/main/java/com/google/sps/model/GmailClientImpl.java

```diff
@@ -42,13 +48,42 @@ private GmailClientImpl(Credential credential) {
             .build();
   }
 
+  /**
+   * List the messages in a user's Gmail account that match the passed query. Results are returned
+   * in batches of 100 from the Gmail API, though a next page token is included if another batch
+   * still exists. This method will use these tokens to get all of the emails that match the query,
+   * even if that is in excess of 100.
+   *
+   * @param query search query to filter which results are returned (see:
+   *     https://support.google.com/mail/answer/7190?hl=en)
+   * @return list of message objects that have an ID and thread ID
+   * @throws IOException if an issue occurs with the gmail service
+   */
   @Override
   public List<Message> listUserMessages(String query) throws IOException {
-    // Null if no messages present. Convert to empty list for ease
-    List<Message> messages =
-        gmailService.users().messages().list("me").setQ(query).execute().getMessages();
-
-    return messages != null ? messages : new ArrayList<>();
+    List<Message> userMessages = new ArrayList<>();
+    List<Message> newBatchUserMessages;
```

Comment:
You can move this variable definition inside the do-while loop.

Reply:
Agreed! Done!